Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h

mikaylagawarecki opened the pull request:
This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #166996
* #166995

mikaylagawarecki committed:
Add version linter for files that include torch/csrc/stable/c/shim.h

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files that include torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files that include torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files that include torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files that include torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files that include torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

janeyx99 commented on tools/linter/adapters/stable_shim_usage_linter.py:
```diff
@@ -0,0 +1,385 @@
+        self.preprocessor_stack: list[tuple[bool, tuple[int, int] | None]] = []
+
+        # Current version requirement (if inside a version block)
+        self.current_version: tuple[int, int] | None = None
```
nit: maybe call this version_of_block cuz current_version got conflated for me with the current torch version (like 2.10)
```suggestion
        self.version_of_block: tuple[int, int] | None = None
```

janeyx99 commented on tools/linter/adapters/stable_shim_usage_linter.py:
```diff
@@ -0,0 +1,385 @@
+                            replacement=None,
+                            description=(
+                                f"Call to versioned shim function '{func_name}' is not wrapped "
+                                f"in a TORCH_FEATURE_VERSION block. This function requires at least:\n"
```
phrasing is a bit weird, but thats ok

janeyx99 reviewed (approved):
some nits + requests for clarity, but overall looks good! thanks for the tests and example files

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Add version linter for files in torch/csrc/stable that use torch/csrc/stable/c/shim.h"

This PR ensures that files in `torch/csrc/stable` that include from `torch/csrc/stable/shim.h` properly gate calls to the stable shim APIs with the appropriate `TORCH_FEATURE_VERSION` macros

### Examples

1. Commenting out the version guard in

https://github.com/pytorch/pytorch/blob/a64c7d740428010d700b4bcd395af8a7b2d5c21f/torch/csrc/stable/ops.h#L265-L271

```
Error (STABLE_SHIM_USAGE) unversioned-shim-call
    Call to versioned shim function 'torch_call_dispatcher' is not wrapped in
    a TORCH_FEATURE_VERSION block. This function requires:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |  // #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |  // #else
        269  |  //   TORCH_ERROR_CODE_CHECK(
 ```
 
2. Changing the first line to say  #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_11_0
 
 Doesn't error because this is still valid!
 
 3. Change that line to #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
 
 ```
 Error (STABLE_SHIM_USAGE) insufficient-version-for-shim-call
    Call to 'torch_call_dispatcher' is wrapped in TORCH_VERSION_2_9_0, but
    this function requires at least TORCH_VERSION_2_10_0. The version guard
    must be at least the required version:
    #if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
      // ... your code calling torch_call_dispatcher ...
    #endif // TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

        263  |      torch::stable::detail::from(self),
        264  |      torch::stable::detail::from(std::nullopt)};
        265  |#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_9_0
    >>> 266  |  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
        267  |      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
        268  |#else
        269  |  TORCH_ERROR_CODE_CHECK(
        
```

4. Nested ifdefs 

```
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0

// We expect this to be the stable version of the clone op. We will
// add optional memory_format kwarg support in the future.
inline torch::stable::Tensor clone(const torch::stable::Tensor& self) {
  const auto num_args = 2;
  std::array<StableIValue, num_args> stack{
      torch::stable::detail::from(self),
      torch::stable::detail::from(std::nullopt)};
#if TORCH_FEATURE_VERSION >= TORCH_VERSION_2_10_0
#endif
#ifdef FOO
  TORCH_ERROR_CODE_CHECK(torch_call_dispatcher(
      "aten::clone", "", stack.data(), TORCH_ABI_VERSION));
#else
  TORCH_ERROR_CODE_CHECK(
      aoti_torch_call_dispatcher("aten::clone", "", stack.data()));
#endif
  return torch::stable::detail::to<torch::stable::Tensor>(stack[0]);
}

#endif
```

No error!
 
 




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge -i

mikaylagawarecki commented:
@pytorchbot merge -i
